federation/pkg/federation-controller/util/federated_informer.go

```diff
@@ -97,8 +100,17 @@ type FederatedInformer interface {
 // framework.DeletionHandlingMetaNamespaceKeyFunc as a keying function.
 type TargetInformerFactory func(*federation_api.Cluster, federation_release_1_4.Interface) (cache.Store, framework.ControllerInterface)
 
+type ClusterLifecycleHandler func(*federation_api.Cluster, FederationView, FederatedReadOnlyStore)
```

Comment:
To see what other clusters are there. Honestly i don't see any direct need in namespace/RS controller but i have a feeling it may be needed. Anyway user can ignore it if not needed.